spec/features/categories_spec.rb

```diff
@@ -8,12 +8,12 @@
 
   let(:product) { Product.first }
   context "returns the Category selected", js: true do
-    it "returns the correct element" do
-      visit product_path(product)
+    it "returns the correct element", :correct_element do
+      visit category_path(product)
       expect(page).to have_css(".category-span", visible: true)
       page.find(".category-span").hover
       expect(page).to have_css('#category-dropdown', visible: true)
-      click_link(product.category.name)
+      first(:link, product.category.name).click
```

Comment:
**Nitpick - Test**
I like that you have begun testing @andela-anwocha. 👍 

However, in testing there are four phases:
- setup
- exercise
- verify
- teardown

If you are unable to strictly follow the phases in this example then it means that you need to split the example into several examples in order to adhere to these phases.

Check out this resource and make necessary adjustments.
https://robots.thoughtbot.com/four-phase-test